src/DynamoCoreWpf/ViewModels/Search/NodeSearchElementViewModel.cs

```diff
@@ -323,35 +337,143 @@ protected virtual bool CanCreateAndConnectToPort(object parameter)
         {
             // Do not auto connect code block node since default code block node do not have output port
             if (Model.CreationName.Contains("Code Block")) return false;
+            
+            // Avoid triggering auto complete for an already connected input.
+            PortModel portModel = parameter as PortModel;
+            return parameter as PortModel != null ? portModel.CanAutoCompleteInput() : true;
+        }
+
+        private Rect2D GetNodesBoundingBox(IEnumerable<NodeModel> nodes)
```

Comment:
Looking great! Thank you for splitting these up. Would you add some comments to these new functions?